### PR TITLE
Modify CONTRIBUTING guidelines based in-person discussions

### PR DESCRIPTION
**Pull request by willbarton**
Modifies the CONTRIBUTING.md file slightly to align with the front-end standards, back-end discussions, our team discussion in-person last week, and our "Building and Deploying" working document.

## Additions

- Adds some additional text about contributing to this repository.

## Review

@cfpb/cfgov-platform 

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,8 +25,16 @@ guidance below.
 ## Changing the code-base
 
 Generally speaking, you should fork this repository, make changes in your
-own fork, and then submit a pull-request. All new code should have associated unit
```
Comment (Scotchester) [inline CONTRIBUTING.md]: Interesting. Hasn't branching in cfgov-refresh been the standard for quite a while now? Does this signal an intention to go back to a fork-first model?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,8 +25,16 @@ guidance below.
...
-Additionally, the code should follow any stylistic and architectural guidelines 
-prescribed by the project. In the absence of such guidelines, mimic the styles
-and patterns in the existing code-base.
+own fork, and then submit a pull-request. For timely code reviews,
+please tag @cfpb/cfgov-backends and @cfpb/cfgov-frontends as appropriate
+for your changes.
+
+All new code should have associated unit tests and/or functional tests that 
+validate implemented features and the presence or lack of defects. The
+overall test coverage of the code-base should not decrease.
```
Comment (Scotchester) [inline CONTRIBUTING.md]: Codebase is one word, no?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,8 +25,16 @@ guidance below.
...
+
+Python code is expected to follow PEP8 and 
+[not commit atrocities](https://www.youtube.com/watch?v=wf-BqAjZb8M). 
+JavaScript should follow our 
+[JavaScript style guide](https://github.com/cfpb/front-end/blob/master/javascript.md). 
+When in doubt, mimic the styles and patterns in the existing code-base.
```
Comment (Scotchester) [inline CONTRIBUTING.md]: No mention of our other front-end language standards?

File: CONTRIBUTING.md (hunk 1 above)
Comment (willbarton) [inline CONTRIBUTING.md]: I don't think we've had any standardization around that though, have we? 

I mean, clearly the CONTRIBUTING file hasn't reflected that practice :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (willbarton) [inline CONTRIBUTING.md]: I'll happily change it. I simply kept the hyphenation from the previous revision.

Comment (Scotchester) [inline CONTRIBUTING.md]: Me thinks it's time to correct it :) Also on L40

File: CONTRIBUTING.md (hunk 1 above)
Comment (Scotchester) [inline CONTRIBUTING.md]: Well, I suppose the Platform team (which is largely different than the V1 team) should make an affirmative decision about what it prefers.

**Comment by Scotchester** [comment]
:+1:

File: CONTRIBUTING.md (hunk 1 above)
Comment (willbarton) [inline CONTRIBUTING.md]: I think for external contributions, we definitely want a fork. For internal (CFPB) contributors, I think there's a lot more fuzz. We have branching and merging guidelines in our working "building for cf.gov" documentation.

**Comment by grapesmoker** [comment]
👍

**Pull request merged**